File: src/test/samples/modifierSample.cs
// Modifier extraction stress-test sample.
// Each method is preceded by comments containing misleading visibility keywords.

#region Private Helpers

/// <summary>
/// Retrieve the private m_connections dictionary from a running server instance.
/// </summary>
public static void PublicAfterPrivateComment() {}

// Generic private-field access helpers for white-box testing.
public static void PublicAfterPrivateLineComment() {}

/* This wraps a private implementation detail. */
public void PublicAfterBlockComment() {}

protected void ProtectedMethod() {}

private void PrivateMethod() {}

internal void InternalMethod() {}

protected internal void ProtectedInternalMethod() {}

#endregion

#region Static Members

public static void StaticMethod() {}

public readonly int ReadOnlyField = 42;

public const string ConstField = "test";

#endregion

File: src/test/samples/modifierSample.ts
// Modifier extraction stress-test sample (TypeScript).
//
// The TypeScript language server is guaranteed present in the test host, so
// `executeDocumentSymbolProvider` returns real symbols for this file. Each
// member is preceded by comments (and lives inside #region blocks) containing
// misleading visibility keywords; the extractor must ignore those and read the
// actual declaration.

// #region Private Helpers

export class ModifierShowcase {
  /**
   * Retrieve the private m_connections map from a running server instance.
   */
  public static publicAfterPrivateComment(): void {}

  // Generic private-field access helpers for white-box testing.
  public static publicAfterPrivateLineComment(): void {}

  /* This wraps a private implementation detail. */
  public publicAfterBlockComment(): void {}

  protected protectedMethod(): void {}

  private privateMethod(): void {}

  // #region Static Members

  public static staticMethod(): void {}

  public readonly readOnlyField: number = 42;

  // #endregion

  public async asyncMethod(): Promise<void> {}
}

// #endregion
